docs/DevGuide/Protocols.md

```diff
@@ -48,26 +64,40 @@ has explicitly defined (and documented!) the interface they expect. And the
 developer who consumes the protocol by writing classes that conform to it knows
 exactly what needs to be implemented.
 
+Note that the `conforms_to` metafunction is SFINAE-friendly, so you can also
+use it like this:
+
+\snippet Utilities/Test_ProtocolHelpers.cpp protocol_sfinae
+
 We typically define protocols in a file named `Protocols.hpp` and within a
 `protocols` namespace, similar to how we write \ref DataBoxTagsGroup "tags" in a
 `Tags.hpp` file and within a `Tags` namespace. The file should be placed in the
-in the directory associated with the code that depends on classes conforming to
-the protocols. For example, the protocol `Named` in the example above would be
+directory associated with the code that depends on classes conforming to the
+protocols. For example, the protocol `Named` in the example above would be
 placed in directory that also has the `greet` function.
 
-## Protocol users: Testing protocol conformance
+# Protocol users: Testing protocol conformance {#protocols_testing_conformance}
 
 Any class that indicates it conforms to the protocol must test that it actually
 does using the `test_protocol_conformance` metafunction from
 `tests/Unit/ProtocolTestHelpers.hpp`:
 
 \snippet Utilities/Test_ProtocolHelpers.cpp test_protocol_conformance
 
-## Protocol authors: Protocols should not be templates
+# Protocol authors: Protocols must be unary type traits {#protocols_author}
 
-When you author a new protocol, keep in mind that protocols should typically not
-be templates. Instead of adding template parameters to the protocol, make them
-part of your protocol instead. The reason for this guideline is that protocols
+When you author a new protocol, keep in mind that protocols must be unary type
+traits. This means they take a single template parameter (typically named
+`ConformingType`) and derive off `std::true_type` or `std::false_type` depending
+on whether the `ConformingType` fullfills the protocol's requirements. Make sure
+to implement the protocol in a SFINAE-friendly way. You may find the macros in
+`Utilities/TypeTraits.hpp` useful. For example, we use `CREATE_IS_CALLABLE` in
+the protocols above for testing the existence and return type of a member
+function.
+
+Occasionally, You might be tempted to add additional template parameters to the
+protocol. In those situations, make the additional parameters part of your
+protocol instead. The reason for this guideline is that protocols
 will always be used as base classes. Therefore, any template parameters of the
```

Comment:
"be used as base classes" -> "be used as unary type traits when inheriting from `ConformsTo<Protocol>`"